File: Arrays/Contains Duplicate/containsDuplicate.ts
// class Solution {
//     containsDuplicate(nums: number[]): boolean {
//       const setNums = new Set(nums);
//       return setNums.size !== nums.length;
//     }
//   }
  
//   class SolutionTwo {
//     containsDuplicate(nums: number[]): boolean {
//       return new Set(nums).size !== nums.length;
//     }
//   }
  
//   const array: number[][] = [[1, 2, 3, 4, 4], [2, 3, 4, 5], [], [3]];
  
//   for (const element of array) {
//     const solution = new Solution();
//     const solutionTwo = new SolutionTwo();
//     console.log(element);
//     console.log(solution.containsDuplicate(element));
//     console.log(solutionTwo.containsDuplicate(element));
//   }

File: Arrays/Contains Duplicate/containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        setNums = len(set(nums))
        if len(nums) != setNums:
            return True
        return False


class SolutionTwo:
    def containsDuplicate(self, nums: list[int]) -> bool:
        return len(set(nums)) != len(nums)


array = [[1, 2, 3, 4, 4], [2, 3, 4, 5], [], [3]]
for element in array:
    solution = Solution()
    solutionTwo = SolutionTwo()
    print(element)
    print(solution.containsDuplicate(element))
    print(solutionTwo.containsDuplicate(element))
